socket warnings

When compiled with gcc on linux:

```
../v7.c: In function 'Socket_connect':
../v7.c: In function 'Socket_connect':
../v7.c:13816:63: warning: comparison is always false due to limited range of data type [-Wtype-limits]
       (port = Socket_get_port(v7, v7_array_get(v7, args, 1))) < 0 ||
                                                               ^
../v7.c: In function 'Socket_sendto':
../v7.c:13872:63: warning: comparison is always false due to limited range of data type [-Wtype-limits]
       (port = Socket_get_port(v7, v7_array_get(v7, args, 1))) < 0 ||
                                                               ^
../v7.c:13816:63: warning: comparison is always false due to limited range of data type [-Wtype-limits]
       (port = Socket_get_port(v7, v7_array_get(v7, args, 1))) < 0 ||
                                                               ^
../v7.c: In function 'Socket_sendto':
../v7.c:13872:63: warning: comparison is always false due to limited range of data type [-Wtype-limits]
       (port = Socket_get_port(v7, v7_array_get(v7, args, 1))) < 0 ||
                                                               ^
../src/socket.c: In function 'Socket_connect':
../src/socket.c:418:63: warning: comparison is always false due to limited range of data type [-Wtype-limits]
       (port = Socket_get_port(v7, v7_array_get(v7, args, 1))) < 0 ||
                                                               ^
../src/socket.c: In function 'Socket_sendto':
../src/socket.c:474:63: warning: comparison is always false due to limited range of data type [-Wtype-limits]
       (port = Socket_get_port(v7, v7_array_get(v7, args, 1))) < 0 ||
                                                               ^
../src/socket.c: In function 'v7_val_t Socket_connect(v7*, val_t, val_t)':
../src/socket.c:418:63: warning: comparison is always false due to limited range of data type [-Wtype-limits]
       (port = Socket_get_port(v7, v7_array_get(v7, args, 1))) < 0 ||
                                                               ^
../src/socket.c: In function 'v7_val_t Socket_sendto(v7*, val_t, val_t)':
../src/socket.c:474:63: warning: comparison is always false due to limited range of data type [-Wtype-limits]
       (port = Socket_get_port(v7, v7_array_get(v7, args, 1))) < 0 ||
```

--- a/src/socket.c
+++ b/src/socket.c
@@ -264,7 +264,7 @@ static int Socket_get_addrCstr(struct v7 *v7, val_t addr_val, char *buf,
  */
 static v7_val_t Socket_bind(struct v7 *v7, val_t this_obj, val_t args) {
   struct socket_internal *si = Socket_get_si(v7, this_obj);
-  int port;
+  int32_t port;
   union socket_address sa;
   int sa_size;
   char addr[ADDRESS_BUF_SIZE] = {0};
@@ -407,7 +407,7 @@ static v7_val_t Socket_connect(struct v7 *v7, val_t this_obj, val_t args) {
   char addr[ADDRESS_BUF_SIZE] = {0};
   union socket_address sa;
   int sa_size;
-  uint16_t port;
+  int32_t port;
 
   if (si == NULL) {
     return Socket_set_last_error(v7, this_obj, ERR_SOCKET_CLOSED);
@@ -457,7 +457,7 @@ static v7_val_t Socket_close(struct v7 *v7, val_t this_obj, val_t args) {
 static v7_val_t Socket_sendto(struct v7 *v7, val_t this_obj, val_t args) {
   struct socket_internal *si = Socket_get_si(v7, this_obj);
   char addr[ADDRESS_BUF_SIZE] = {0};
-  uint16_t port;
+  int32_t port;
   union socket_address sa;
   int sa_size;
   uint8_t *buf = NULL, *ptr;

--- a/tests/ecma_report.txt
+++ b/tests/ecma_report.txt
@@ -11174,7 +11174,7 @@ $I: [{"message":"#1: Incorrect value of Date"}]
 11066	PASS ch15/15.9/15.9.5/15.9.5.43/15.9.5.43-0-16.js (tail -c +10078357 tests/ecmac.db|head -c 596)
 11067	FAIL ch15/15.9/15.9.5/15.9.5.43/15.9.5.43-0-2.js (tail -c +10078954 tests/ecmac.db|head -c 335): [{"message":"Test case returned non-true value!"}]
 11068	PASS ch15/15.9/15.9.5/15.9.5.43/15.9.5.43-0-3.js (tail -c +10079290 tests/ecmac.db|head -c 326)
-11069	FAIL ch15/15.9/15.9.5/15.9.5.43/15.9.5.43-0-4.js (tail -c +10079617 tests/ecmac.db|head -c 535): [{"message":"Test case returned non-true value!"}]
+11069	PASS ch15/15.9/15.9.5/15.9.5.43/15.9.5.43-0-4.js (tail -c +10079617 tests/ecmac.db|head -c 535)
 11070	FAIL ch15/15.9/15.9.5/15.9.5.43/15.9.5.43-0-5.js (tail -c +10080153 tests/ecmac.db|head -c 383): [{"message":"Test case returned non-true value!"}]
 11071	PASS ch15/15.9/15.9.5/15.9.5.43/15.9.5.43-0-6.js (tail -c +10080537 tests/ecmac.db|head -c 483)
 11072	PASS ch15/15.9/15.9.5/15.9.5.43/15.9.5.43-0-7.js (tail -c +10081021 tests/ecmac.db|head -c 463)

--- a/v7.c
+++ b/v7.c
@@ -13687,7 +13687,7 @@ static int Socket_get_addrCstr(struct v7 *v7, val_t addr_val, char *buf,
  */
 static v7_val_t Socket_bind(struct v7 *v7, val_t this_obj, val_t args) {
   struct socket_internal *si = Socket_get_si(v7, this_obj);
-  int port;
+  int32_t port;
   union socket_address sa;
   int sa_size;
   char addr[ADDRESS_BUF_SIZE] = {0};
@@ -13830,7 +13830,7 @@ static v7_val_t Socket_connect(struct v7 *v7, val_t this_obj, val_t args) {
   char addr[ADDRESS_BUF_SIZE] = {0};
   union socket_address sa;
   int sa_size;
-  uint16_t port;
+  int32_t port;
 
   if (si == NULL) {
     return Socket_set_last_error(v7, this_obj, ERR_SOCKET_CLOSED);
@@ -13880,7 +13880,7 @@ static v7_val_t Socket_close(struct v7 *v7, val_t this_obj, val_t args) {
 static v7_val_t Socket_sendto(struct v7 *v7, val_t this_obj, val_t args) {
   struct socket_internal *si = Socket_get_si(v7, this_obj);
   char addr[ADDRESS_BUF_SIZE] = {0};
-  uint16_t port;
+  int32_t port;
   union socket_address sa;
   int sa_size;
   uint8_t *buf = NULL, *ptr;

